File: packages/transform/src/features/misc/MiscValidatePruneTransformer.ts
import { MiscValidatePruneProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscValidatePruneTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "misc.validatePrune",
      write: MiscValidatePruneProgrammer.write,
    });
}

File: packages/transform/src/features/misc/MiscValidatePruneTransformer.go
package misc

import (
	coremisc "github.com/samchon/typia/packages/core/src/programmers/misc"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var MiscValidatePruneTransformer = miscValidatePruneTransformerNamespace{}

type miscValidatePruneTransformerNamespace struct{}

func (miscValidatePruneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	prune := coremisc.MiscPruneProgrammer.Write(call.TypeText)
	return "((input) => ({ success: true, data: " + prune + "(input) }))"
}
